src/granite_orm/querying.cr

```diff
@@ -45,6 +45,16 @@ module Granite::ORM::Querying
     return rows
   end
 
+  # First adds a `LIMIT 1` clause to the query and returns the first result
+  def first(clause = "", params = [] of DB::Any)
+    results = all([clause.strip, "LIMIT 1"].join(" "), params)
+    if results.size > 0
+      results.first
+    else
+      nil
+    end
```

Comment:
How about `results[0]?`?

Reply:
Is this a stylistic choice or is there a performance difference? Just curious

Reply:
We don't need to check size here because `#[]?` did it.

Reply:
Oh I just found a better choice which is `#.first?`.

Reply:
good call, I will use that